packages/web-frontend/src/sagas.js

```diff
@@ -690,11 +694,28 @@ function* fetchDashboard(action) {
   const { organisationUnitCode } = action.organisationUnit;
   const state = yield select();
   const projectCode = selectCurrentProjectCode(state);
+  const project = selectCurrentProject(state);
   const currentDashboardCode = selectCurrentDashboardGroupCodeFromLocation(state);
   const requestResourceUrl = `dashboard?organisationUnitCode=${organisationUnitCode}&projectCode=${projectCode}`;
 
   try {
     const dashboard = yield call(request, requestResourceUrl, fetchDashboardError);
+
+    // If there is no dashboard code defined, assign the default if it is valid for the user
+    if (!selectIsDashboardGroupCodeDefined(state)) {
+      const currentDashboardGroupCode = project.dashboardGroupName;
+
+      if (dashboard[currentDashboardGroupCode]) {
+        yield put(setDashboardGroup(project.dashboardGroupName));
+      }
+    }
+
+    // Check if the user has permission to view the dashboard
+    if (currentDashboardCode && !(currentDashboardCode in dashboard)) {
+      yield call(handleInvalidPermission, { projectCode });
+      return;
+    }
```

Comment:
Would this mean that if a user accidentally mistypes or uses a broken link to see a dashboard they will get prompted with the insufficient permissions dialog? Or will such a request throw an error?